src/serialization/AOF/rai_aof_rewrite.c

```diff
@@ -75,29 +62,54 @@ void RAI_AOFRewriteModel(RedisModuleIO *aof, RedisModuleString *key, void *value
 
     const char *backendstr = RAI_BackendName(model->backend);
 
-    RedisModule_EmitAOF(aof, "AI.MODELSET", "slccclclcvcvcv", key, backendstr, model->devicestr,
-                        model->tag, "BATCHSIZE", model->opts.batchsize, "MINBATCHSIZE",
-                        model->opts.minbatchsize, "INPUTS", inputs_, model->ninputs, "OUTPUTS",
-                        outputs_, model->noutputs, "BLOB", buffers_, n_chunks);
+    if (model->backend != RAI_BACKEND_TENSORFLOW) {
+
+        RedisModule_EmitAOF(aof, "AI.MODELSTORE", "scccsclclclcv", key, backendstr,
+                            model->devicestr, "TAG", model->tag, "BATCHSIZE", model->opts.batchsize,
+                            "MINBATCHSIZE", model->opts.minbatchsize, "MINBATCHTIMEOUT",
+                            model->opts.minbatchtimeout, "BLOB", buffers_, n_chunks);
+    } else {
+        // For TF backend, the command should contain INPUTS and OUTPUTS names.
+        // Create RedisModuleString* arrays from the char* arrays, so we can send a proper vector
+        // to RedisModule_EmitAOF.
+        RedisModuleString **inputs_ = array_new(RedisModuleString *, model->ninputs);
+        RedisModuleString **outputs_ = array_new(RedisModuleString *, model->noutputs);
```

Comment:
new on the stack?

Reply:
I'll change it (it will require using an arbitrary const as `cap`)